File: user_struct.py
import json
import random
import string

# Определение класса, аналогичного структуре User в Rust
class User:
    def __init__(self, name, age):
        self.name = name
        self.age = age

# Создание экземпляра класса User
user = User("Алексей", 30)

# Генерация случайного имени файла
random_filename = "user_data_" + ''.join(random.choices(string.ascii_letters + string.digits, k=10)) + ".txt"

# Сохранение данных в файл в формате JSON
with open(random_filename, 'w', encoding='utf-8') as f:
    json.dump({"name": user.name, "age": user.age}, f, ensure_ascii=False, indent=4)

print(f"Структура User сохранена в файл: {random_filename}")
print(f"Данные: имя - {user.name}, возраст - {user.age}")

File: user_struct.rs
// Определение структуры пользователя
struct User {
    name: String,
    age: u32,
}

// Пример использования структуры и сохранения в файл
// (код ниже демонстрирует, как это будет выглядеть в Rust-приложении)
/*
use std::fs::File;
use std::io::Write;
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
struct User {
    name: String,
    age: u32,
}

fn main() -> std::io::Result<()> {
    // Создаем экземпляр структуры User
    let user = User {
        name: String::from("Иван"),
        age: 30,
    };

    // Генерируем случайное имя файла (в реальном приложении использовалась бы библиотека rand)
    let random_filename = format!("user_data_{}.json", rand::random::<u32>());
    
    // Сериализуем структуру в JSON и записываем в файл
    let json_data = serde_json::to_string(&user)?;
    let mut file = File::create(&random_filename)?;
    file.write_all(json_data.as_bytes())?;
    
    println!("Структура пользователя сохранена в файл: {}", random_filename);
    
    Ok(())
}
*/

// Просто создаем экземпляр структуры для демонстрации
fn main() {
    let user = User {
        name: String::from("Иван"),
        age: 30,
    };
    
    println!("Создан пользователь: {}, возраст: {}", user.name, user.age);
}